sys/shell/cmds/nanocoap_vfs.c

```diff
@@ -52,16 +52,30 @@ static bool _is_dir(const char *url)
 
 static int _resource_cb(char *entry, void *ctx)
 {
-    (void)ctx;
+    bool *too_long = ctx;
+    char *start, *end;
 
-    char *start = strchr(entry, '<');
-    if (start) {
-        char *end = strchr(entry, '>');
+    if (*too_long) {
+        goto find_end;
+    }
+
+    start = strchr(entry, '<');
+    if (start == NULL) {
+        return 0;
+    }
+    entry = start + 1;
+
+find_end:
+    end = strchr(entry, '>');
+    if (end == NULL) {
+        *too_long = true;
+        printf("%s", entry);
```

Comment:
Is `entry` `\0` terminated here?

Reply:
Yes, `nanocoap_link_format_get_url()` will take care of this.